radix-engine-interface/src/model/resource/non_fungible_address.rs

```diff
@@ -91,6 +93,41 @@ impl NonFungibleAddress {
         vec.append(&mut other_vec);
         vec
     }
+
+    /// Returns canonical representation of this NonFungibleAddress.
+    pub fn to_canonical_string(&self, bech32_encoder: &Bech32Encoder) -> String {
+        format!(
+            "{}:{}",
+            bech32_encoder.encode_resource_address_to_string(&self.resource_address),
+            self.non_fungible_id.to_simple_string()
+        )
+    }
+
+    /// Converts canonical representation to NonFungibleAddress.
+    pub fn try_from_canonical_string(
+        bech32_decoder: &Bech32Decoder,
+        id_type: NonFungibleIdType,
+        s: &str,
+    ) -> Result<Self, ParseNonFungibleAddressError> {
+        if let Some(idx) = s.find(":") {
+            if idx > 0 && idx < s.len() - 1 {
+                if let Ok(raddr) = bech32_decoder.validate_and_decode_resource_address(&s[0..idx]) {
+                    if let Ok(nfid) = NonFungibleId::try_from_simple_string(id_type, &s[idx + 1..])
+                    {
+                        Ok(NonFungibleAddress::new(raddr, nfid))
+                    } else {
+                        Err(ParseNonFungibleAddressError::InvalidNonFungibleId)
+                    }
+                } else {
+                    Err(ParseNonFungibleAddressError::InvalidResourceAddress)
+                }
+            } else {
+                Err(ParseNonFungibleAddressError::InvalidLength(s.len()))
```

Comment:
Minor: Misleading error message. I would use `.split().collect()` and give `InvalidNumberOfParts` error.

Reply:
done